src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -188,8 +188,14 @@ private void declareNamespace(String namespace, TypedVar symbol, boolean isDefau
       for (String property : objType.getPropertyNames()) {
         desiredSymbols.add(symbol.getName() + "." + property);
       }
-      // These will be emitted in their own default-exported namespace
-      desiredSymbols.removeAll(provides);
+      // Any provides have their own namespace and should not be emitted in this namespace.
```

Comment:
`Any provides (and their implicit parent namespaces)` to explain why we not only handle a provide like `a.b.c.` but also `a.b` and `a`.